lib/util.js

```diff
@@ -22,6 +22,17 @@ const fs = require('fs');
  */
 function createReadStream(path, options) {
     return new Promise((resolve, reject) => {
+        if(typeof options === 'string' || options instanceof String){
+            const encoding = options;
+            options = {};
+            options.encoding = encoding;
+        }
+        if(typeof options === 'object') { // ensure flags are here                
+            options.flags = options.flags || 'r';
+            options.encoding = options.encoding || 'utf8';
+            options.mode = options.mode || 0o444;
```

Comment:
I think this could be changed to this more readable form:

```suggestion
            options = Object.assign({
                flags: 'r',
                encoding: 'utf8',
                mode: 0o444
            }, options);
```

Reply:
FYI, our eslint requires using spread ({...bleh}).

Reply:
Not that I know. Someone must have added it. It should be a hint/warning, not an error.